ambari-server/src/main/java/org/apache/ambari/server/state/ServiceInfo.java

```diff
@@ -59,7 +59,7 @@
 
 @XmlAccessorType(XmlAccessType.FIELD)
 @JsonFilter("propertiesfilter")
-public class ServiceInfo implements Validable {
+public class ServiceInfo implements Validable, Cloneable {
```

Comment:
Hmmm - I always disliked `Cloneable` b/c it's very easy to miss changes made to the class. What about something like `SerializationUtils#clone` from `commons.lang`?

Reply:
Hmm, but this will require all stack objects to be serializable. For now I will stick with what I have and if required will look at this later.